In [1]:
import os

print(os.path.exists("data/pdbbind2016_core_test.hdf"))

True


In [2]:
19*48*48*48 # 2101248 ~ 2M data points for each complex

2101248

In [3]:
48*48*48 # 110592 ~ 100K data points for each feature

110592

In [4]:
import h5py

import numpy as np
import pandas as pd

filename = "data/pdbbind2016_core_test.hdf"

with h5py.File(filename, "r") as pdb:
    # Print all root level object names (aka keys) 
    # these can be group or dataset names 
    print(f"Keys, {len(pdb.keys())}: {pdb.keys()}")

    print("-------------------------")
    
    # understand 1bzc data
    complex = "1a30"

    print(f"Compex : {complex}, Complex Data Type: {type(pdb[complex])}") 
    print("-------------------------")

    # If a_group_key is a group name, 
    # this gets the object names in the group and returns as a list
    data1 = list(pdb[complex])


    # If a_group_key is a dataset name, 
    # this gets the dataset values and returns as a list
    # data = list(f[a_group_key])
    print("d1", type(data1), len(data1), data1[:5])


    # If a_group_key is a group name, 
    # this gets the object names in the group and returns as a list
    data2 = list(pdb[complex][data1[0]])


    # If a_group_key is a dataset name, 
    # this gets the dataset values and returns as a list
    # data = list(f[a_group_key])
    print("d2", type(data2), len(data2), data2[:5])

    # If a_group_key is a group name, 
    # this gets the object names in the group and returns as a list
    data3 = list(pdb[complex][data1[0]][data2[0]])


    # If a_group_key is a dataset name, 
    # this gets the dataset values and returns as a list
    # data = list(f[a_group_key])
    print("d3", type(data3), len(data3), data3[:5])

    # If a_group_key is a group name, 
    # this gets the object names in the group and returns as a list
    data4 = list(pdb[complex][data1[0]][data2[0]][data3[0]])


    # If a_group_key is a dataset name, 
    # this gets the dataset values and returns as a list
    # data = list(f[a_group_key])
    print("d4", type(data4), len(data4), data4[:5])

    # If a_group_key is a group name, 
    # this gets the object names in the group and returns as a list
    data5 = list(pdb[complex][data1[0]][data2[0]][data3[0]][data4[0]])


    # If a_group_key is a dataset name, 
    # this gets the dataset values and returns as a list
    # data = list(f[a_group_key])
    print("d5", type(data5), len(data5))

    data5_np = np.array(data5)
    print("d5-np", type(data5), data5_np.shape,len(data5_np[1]), data5_np[1], )




Keys, 285: <KeysViewHDF5 ['1a30', '1bcu', '1bzc', '1c5z', '1e66', '1eby', '1g2k', '1gpk', '1gpn', '1h22', '1h23', '1k1i', '1lpg', '1mq6', '1nc1', '1nc3', '1nvq', '1o0h', '1o3f', '1o5b', '1owh', '1oyt', '1p1n', '1p1q', '1ps3', '1pxn', '1q8t', '1q8u', '1qf1', '1qkt', '1r5y', '1s38', '1sqa', '1syi', '1u1b', '1uto', '1vso', '1w4o', '1y6r', '1yc1', '1ydr', '1ydt', '1z6e', '1z95', '1z9g', '2al5', '2br1', '2brb', '2c3i', '2cbv', '2cet', '2fvd', '2fxs', '2hb1', '2iwx', '2j78', '2j7h', '2p15', '2p4y', '2pog', '2qbp', '2qbq', '2qbr', '2qe4', '2qnq', '2r9w', '2v00', '2v7a', '2vkm', '2vvn', '2vw5', '2w4x', '2w66', '2wbg', '2wca', '2weg', '2wer', '2wn9', '2wnc', '2wtv', '2wvt', '2x00', '2xb8', '2xbv', '2xdl', '2xii', '2xj7', '2xnb', '2xys', '2y5h', '2yfe', '2yge', '2yki', '2ymd', '2zb1', '2zcq', '2zcr', '2zda', '2zy1', '3acw', '3ag9', '3ao4', '3arp', '3arq', '3aru', '3arv', '3ary', '3b1m', '3b27', '3b5r', '3b65', '3b68', '3bgz', '3bv9', '3cj4', '3coy', '3coz', '3d4z', '3d6q', '3dd0', '3dx1', '3dx2'

### HDF data explained

- 285 complexes
- For each complex, hiearchy is "pybel","processed", "pdbbind", "data"
- data.shape is (396, 22) so possibly they have a way to reduce it to 392 rows, because actual process is ~2M(19*48*48*48) for each complex and ~100K(48*48*48) data points for each feature in complex.
- First 3 attributes are xyz co-ord, rest are features
